src/util/cron.c

```diff
@@ -243,25 +243,16 @@ void Cron_AbortTask(CronTaskHandle t) {
 	CRON_TASK *task = (CRON_TASK *)t;
 
 	pthread_mutex_lock(&cron->mutex);
-	if(Heap_contains_item(cron->tasks, task)) {
-		// as long as we're holding the cron's mutex it is safe to access task
-
-		CRON_TASK_STATE state = task->state;
-		if(state != TASK_COMPLETED && state != TASK_ABORT) {
-			// try marking task as aborted
-			bool abort = CRON_TaskAdvanceState(task, TASK_PENDING, TASK_ABORT);
-			if(!abort) {
-				// task is executing, wait for it to finish
-				// shouldn't happen often and shouldn't take long
-				volatile CRON_TASK *task_pending = task;
-				while(task_pending->state != TASK_COMPLETED);
-			}
-		}
+	int contains = Heap_contains_item(cron->tasks, task);
+	pthread_mutex_unlock(&cron->mutex);
+	if(!contains) return;
 
-		state = task->state;
-		ASSERT(state == TASK_COMPLETED || state == TASK_ABORT);
+	// try marking task as aborted
+	bool abort = CRON_TaskAdvanceState(task, TASK_PENDING, TASK_ABORT);
+	if(abort) {
+		CRON_RemoveTask(task);
 	}
 
-	pthread_mutex_unlock(&cron->mutex);
+	ASSERT(task->state == TASK_COMPLETED || task->state == TASK_ABORT);
```

Comment:
I don't understand why we can't have the Cron's Run loop remove the items from the heap,
Also I think this ASSERT can fail, e.g. a task is currently executing.